src/components/widget/line/widget-chart-line.component.ts

```diff
@@ -54,47 +56,91 @@ const WidgetChartLineComponent: ng.IComponentOptions = {
-              isFieldRequest = false;
-            }
-            let label = this.parent.widget.chart.labels ? this.parent.widget.chart.labels[idx] : '';
-            if (!label || (value.metadata && value.metadata[bucket.name])) {
-              label = value.metadata[bucket.name].name;
+        i++;
+        let lineColor = ChartService.colorByBucket[i % ChartService.colorByBucket.length];
+        let bgColor = ChartService.bgColorByBucket[i % ChartService.bgColorByBucket.length];
+        let label = this.parent.widget.chart.labels ? this.parent.widget.chart.labels[idx] : '';
+
+        if (value.buckets.length === 0) {
+          let bucketCount = (data.timestamp.to - data.timestamp.from) / data.timestamp.interval;
+
+          this.series.values.push({
+            name: label,
+            data: Array(bucketCount).fill(0),
+            color: lineColor,
+            fillColor: bgColor,
+            labelPrefix: label,
+            id: label,
+            visible: true
+          });
+        } else {
+          value.buckets.forEach((bucket) => {
+            if (bucket) {
+              let isFieldRequest = this.parent.widget.chart.request.aggs.split('%3B')[idx].includes('field:');
+              let query = this.parent.widget.chart.request.query;
+              if (bucket.name === '1' || bucket.name.match('^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$')) {
+                isFieldRequest = false;
+              }
+              if (!label || (value.metadata && value.metadata[bucket.name])) {
+                label = value.metadata[bucket.name].name;
+              }
+
+              this.series.values.push({
+                name: isFieldRequest ? bucket.name : label,
+                data: bucket.data, color: lineColor, fillColor: bgColor,
+                labelPrefix: isFieldRequest ? label : '',
+                id: bucket.name,
+                visible: this.parent.widget.chart.selectable && query ? query.includes(bucket.name) : true
+              });
             }
-
-            this.series.values.push({
-              name: isFieldRequest ? bucket.name : label,
-              data: bucket.data, color: lineColor, fillColor: bgColor,
-              labelPrefix: isFieldRequest ? label : '',
-              id: bucket.name,
-              visible: this.parent.widget.chart.selectable && query ? query.includes(bucket.name) : true
-            });
-          }
-        });
+          });
+        }
       });
 
-      let timestamp = data.timestamp;
-
       this.options = {
         labelPrefix: 'HTTP Status',
         pointStart: timestamp.from,
         pointInterval: timestamp.interval,
-        stacking: this.parent.widget.chart.stacked ? 'normal' : null
+        stacking: this.parent.widget.chart.stacked ? 'normal' : null,
+        plotLines: !this.events ? [] : [
+          ...this.events.map(event => {
```

Comment:
you don't need to destructure and rebuild an array as `map` is already returning a new one:
```
plotLines: !this.events ? [] : this.events.map() ...
```

Reply:
as discussed, we will prefer `(this.events || []).map()...`